File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/fetch_lineage_graph_node_neighbors_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class FetchLineageGraphNodeNeighborsDetails(object):
    """
    The information needed to obtain neighbors for the node.
    """

    #: A constant which can be used with the direction property of a FetchLineageGraphNodeNeighborsDetails.
    #: This constant has a value of "UPSTREAM"
    DIRECTION_UPSTREAM = "UPSTREAM"

    #: A constant which can be used with the direction property of a FetchLineageGraphNodeNeighborsDetails.
    #: This constant has a value of "BOTH"
    DIRECTION_BOTH = "BOTH"

    #: A constant which can be used with the direction property of a FetchLineageGraphNodeNeighborsDetails.
    #: This constant has a value of "DOWNSTREAM"
    DIRECTION_DOWNSTREAM = "DOWNSTREAM"

    def __init__(self, **kwargs):
        """
        Initializes a new FetchLineageGraphNodeNeighborsDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param node_id:
            The value to assign to the node_id property of this FetchLineageGraphNodeNeighborsDetails.
        :type node_id: str

        :param direction:
            The value to assign to the direction property of this FetchLineageGraphNodeNeighborsDetails.
            Allowed values for this property are: "UPSTREAM", "BOTH", "DOWNSTREAM"
        :type direction: str

        :param is_with_columns:
            The value to assign to the is_with_columns property of this FetchLineageGraphNodeNeighborsDetails.
        :type is_with_columns: bool

        :param scoped_columns:
            The value to assign to the scoped_columns property of this FetchLineageGraphNodeNeighborsDetails.
        :type scoped_columns: list[str]

        :param filter:
            The value to assign to the filter property of this FetchLineageGraphNodeNeighborsDetails.
        :type filter: oci.aidataplatform_dp.models.LineageGraphNodeNeighborsFilter

        """
        self.swagger_types = {
            'node_id': 'str',
            'direction': 'str',
            'is_with_columns': 'bool',
            'scoped_columns': 'list[str]',
            'filter': 'LineageGraphNodeNeighborsFilter'
        }

        self.attribute_map = {
            'node_id': 'nodeId',
            'direction': 'direction',
            'is_with_columns': 'isWithColumns',
            'scoped_columns': 'scopedColumns',
            'filter': 'filter'
        }

        self._node_id = None
        self._direction = None
        self._is_with_columns = None
        self._scoped_columns = None
        self._filter = None

    @property
    def node_id(self):
        """
        **[Required]** Gets the node_id of this FetchLineageGraphNodeNeighborsDetails.
        ID of the node to fetch the details


        :return: The node_id of this FetchLineageGraphNodeNeighborsDetails.
        :rtype: str
        """
        return self._node_id

    @node_id.setter
    def node_id(self, node_id):
        """
        Sets the node_id of this FetchLineageGraphNodeNeighborsDetails.
        ID of the node to fetch the details


        :param node_id: The node_id of this FetchLineageGraphNodeNeighborsDetails.
        :type: str
        """
        self._node_id = node_id

    @property
    def direction(self):
        """
        **[Required]** Gets the direction of this FetchLineageGraphNodeNeighborsDetails.
        Direction of the lineage returned.

        Allowed values for this property are: "UPSTREAM", "BOTH", "DOWNSTREAM"


        :return: The direction of this FetchLineageGraphNodeNeighborsDetails.
        :rtype: str
        """
        return self._direction

    @direction.setter
    def direction(self, direction):
        """
        Sets the direction of this FetchLineageGraphNodeNeighborsDetails.
        Direction of the lineage returned.


        :param direction: The direction of this FetchLineageGraphNodeNeighborsDetails.
        :type: str
        """
        allowed_values = ["UPSTREAM", "BOTH", "DOWNSTREAM"]
        if not value_allowed_none_or_none_sentinel(direction, allowed_values):
            raise ValueError(
                "Invalid value for `direction`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._direction = direction

    @property
    def is_with_columns(self):
        """
        Gets the is_with_columns of this FetchLineageGraphNodeNeighborsDetails.
        The flag to indicate whether columns also to be included while fetching neighbors. Default is false.


        :return: The is_with_columns of this FetchLineageGraphNodeNeighborsDetails.
        :rtype: bool
        """
        return self._is_with_columns

    @is_with_columns.setter
    def is_with_columns(self, is_with_columns):
        """
        Sets the is_with_columns of this FetchLineageGraphNodeNeighborsDetails.
        The flag to indicate whether columns also to be included while fetching neighbors. Default is false.


        :param is_with_columns: The is_with_columns of this FetchLineageGraphNodeNeighborsDetails.
        :type: bool
        """
        self._is_with_columns = is_with_columns

    @property
    def scoped_columns(self):
        """
        Gets the scoped_columns of this FetchLineageGraphNodeNeighborsDetails.
        List of scoped column names for fetching the Graph Node Neighbors. Applicable only when isWithColumns is true.


        :return: The scoped_columns of this FetchLineageGraphNodeNeighborsDetails.
        :rtype: list[str]
        """
        return self._scoped_columns

    @scoped_columns.setter
    def scoped_columns(self, scoped_columns):
        """
        Sets the scoped_columns of this FetchLineageGraphNodeNeighborsDetails.
        List of scoped column names for fetching the Graph Node Neighbors. Applicable only when isWithColumns is true.


        :param scoped_columns: The scoped_columns of this FetchLineageGraphNodeNeighborsDetails.
        :type: list[str]
        """
        self._scoped_columns = scoped_columns

    @property
    def filter(self):
        """
        Gets the filter of this FetchLineageGraphNodeNeighborsDetails.

        :return: The filter of this FetchLineageGraphNodeNeighborsDetails.
        :rtype: oci.aidataplatform_dp.models.LineageGraphNodeNeighborsFilter
        """
        return self._filter

    @filter.setter
    def filter(self, filter):
        """
        Sets the filter of this FetchLineageGraphNodeNeighborsDetails.

        :param filter: The filter of this FetchLineageGraphNodeNeighborsDetails.
        :type: oci.aidataplatform_dp.models.LineageGraphNodeNeighborsFilter
        """
        self._filter = filter

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/FetchLineageGraphNodeNeighborsDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * The information needed to obtain neighbors for the node.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=FetchLineageGraphNodeNeighborsDetails.Builder.class)

public final class FetchLineageGraphNodeNeighborsDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"nodeId", "direction", "isWithColumns", "scopedColumns", "filter"})
    public FetchLineageGraphNodeNeighborsDetails(String nodeId, LineageDirection direction, Boolean isWithColumns, java.util.List<String> scopedColumns, LineageGraphNodeNeighborsFilter filter) {
        super();
        this.nodeId = nodeId;
        this.direction = direction;
        this.isWithColumns = isWithColumns;
        this.scopedColumns = scopedColumns;
        this.filter = filter;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * ID of the node to fetch the details
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("nodeId")
private String nodeId;

        /**
         * ID of the node to fetch the details
* 
         * @param nodeId the value to set
         * @return this builder
         **/
        

public Builder nodeId(String nodeId) {
    this.nodeId = nodeId;
    return this;
}
            /**
     * Direction of the lineage returned.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("direction")
private LineageDirection direction;

        /**
         * Direction of the lineage returned.
         * @param direction the value to set
         * @return this builder
         **/
        

public Builder direction(LineageDirection direction) {
    this.direction = direction;
    return this;
}
            /**
     * The flag to indicate whether columns also to be included while fetching neighbors. Default is false.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isWithColumns")
private Boolean isWithColumns;

        /**
         * The flag to indicate whether columns also to be included while fetching neighbors. Default is false.
* 
         * @param isWithColumns the value to set
         * @return this builder
         **/
        

public Builder isWithColumns(Boolean isWithColumns) {
    this.isWithColumns = isWithColumns;
    return this;
}
            /**
     * List of scoped column names for fetching the Graph Node Neighbors. Applicable only when isWithColumns is true.
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("scopedColumns")
private java.util.List<String> scopedColumns;

        /**
         * List of scoped column names for fetching the Graph Node Neighbors. Applicable only when isWithColumns is true.
* 
         * @param scopedColumns the value to set
         * @return this builder
         **/
        

public Builder scopedColumns(java.util.List<String> scopedColumns) {
    this.scopedColumns = scopedColumns;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("filter")
private LineageGraphNodeNeighborsFilter filter;



public Builder filter(LineageGraphNodeNeighborsFilter filter) {
    this.filter = filter;
    return this;
}


        public FetchLineageGraphNodeNeighborsDetails build() {
            FetchLineageGraphNodeNeighborsDetails model = new FetchLineageGraphNodeNeighborsDetails(this.nodeId
                , this.direction
                , this.isWithColumns
                , this.scopedColumns
                , this.filter);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(FetchLineageGraphNodeNeighborsDetails model) {
                this.nodeId(model.getNodeId());
    this.direction(model.getDirection());
    this.isWithColumns(model.getIsWithColumns());
    this.scopedColumns(model.getScopedColumns());
    this.filter(model.getFilter());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * ID of the node to fetch the details
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("nodeId")
    private final String nodeId;

        /**
     * ID of the node to fetch the details
* 
     * @return the value
     **/
    
    public String getNodeId() {
        return nodeId;
    }

    
        /**
     * Direction of the lineage returned.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("direction")
    private final LineageDirection direction;

        /**
     * Direction of the lineage returned.
     * @return the value
     **/
    
    public LineageDirection getDirection() {
        return direction;
    }


        /**
     * The flag to indicate whether columns also to be included while fetching neighbors. Default is false.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isWithColumns")
    private final Boolean isWithColumns;

        /**
     * The flag to indicate whether columns also to be included while fetching neighbors. Default is false.
* 
     * @return the value
     **/
    
    public Boolean getIsWithColumns() {
        return isWithColumns;
    }


        /**
     * List of scoped column names for fetching the Graph Node Neighbors. Applicable only when isWithColumns is true.
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("scopedColumns")
    private final java.util.List<String> scopedColumns;

        /**
     * List of scoped column names for fetching the Graph Node Neighbors. Applicable only when isWithColumns is true.
* 
     * @return the value
     **/
    
    public java.util.List<String> getScopedColumns() {
        return scopedColumns;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("filter")
    private final LineageGraphNodeNeighborsFilter filter;

    
    public LineageGraphNodeNeighborsFilter getFilter() {
        return filter;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("FetchLineageGraphNodeNeighborsDetails(");
        sb.append("nodeId=").append(String.valueOf(this.nodeId));
        sb.append(", direction=").append(String.valueOf(this.direction));
        sb.append(", isWithColumns=").append(String.valueOf(this.isWithColumns));
        sb.append(", scopedColumns=").append(String.valueOf(this.scopedColumns));
        sb.append(", filter=").append(String.valueOf(this.filter));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof FetchLineageGraphNodeNeighborsDetails)) {
            return false;
        }

        FetchLineageGraphNodeNeighborsDetails other = (FetchLineageGraphNodeNeighborsDetails) o;
        return java.util.Objects.equals(this.nodeId, other.nodeId) &&
            java.util.Objects.equals(this.direction, other.direction) &&
            java.util.Objects.equals(this.isWithColumns, other.isWithColumns) &&
            java.util.Objects.equals(this.scopedColumns, other.scopedColumns) &&
            java.util.Objects.equals(this.filter, other.filter);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.nodeId == null ? 43 : this.nodeId.hashCode());
        result = (result * PRIME) + (this.direction == null ? 43 : this.direction.hashCode());
        result = (result * PRIME) + (this.isWithColumns == null ? 43 : this.isWithColumns.hashCode());
        result = (result * PRIME) + (this.scopedColumns == null ? 43 : this.scopedColumns.hashCode());
        result = (result * PRIME) + (this.filter == null ? 43 : this.filter.hashCode());
        return result;
    }


}
